pages/library/_viewId.vue

```diff
@@ -157,6 +198,127 @@ export default Vue.extend({
     }
   },
   methods: {
+    async filterMedia() {
+      try {
+        const collectionInfo = await this.$api.items.getItems({
+          uId: this.$auth.user.Id,
+          userId: this.$auth.user.Id,
+          ids: this.$route.params.viewId
+        });
+
+        if (
+          collectionInfo.data.Items &&
+          (collectionInfo.data.Items[0].Type === 'CollectionFolder' ||
+            collectionInfo.data.Items[0].Type === 'Folder')
+        ) {
+          const options: any = {
+            uId: this.$auth.user.Id,
+            userId: this.$auth.user.Id,
+            parentId: this.$route.params.viewId,
+            includeItemTypes: '',
+            recursive: true,
+            sortBy: this.orderMethod,
+            sortOrder: this.sortDirection,
+            fields: this.orderMethod
+          };
+
+          if (collectionInfo.data.Items[0].CollectionType === 'tvshows') {
+            options.includeItemTypes = 'Series';
+          } else if (collectionInfo.data.Items[0].CollectionType === 'movies') {
+            options.includeItemTypes = 'Movie';
+          } else if (collectionInfo.data.Items[0].CollectionType === 'books') {
+            options.includeItemTypes = 'Book';
+          }
+
+          const filterString = this.makeFilterString(
+            this.allFilters.filters.items,
+            ','
+          );
+          options.filters = filterString;
```

Comment:
I don't get why you're fetching these again. You should pass data around instead of fetching it multiple times (You're already passing all of this in the change event for the filter menu, for example, which is bound to this. So one of your parameters would already contain the list).